docker/credentials_helper_test.go

```diff
@@ -2,26 +2,31 @@ package docker
 
 import "testing"
 
-func Test_to2ndLevelDomain(t *testing.T) {
+func Test_registryEquals(t *testing.T) {
 	tests := []struct {
-		name   string
-		rawurl string
-		want   string
+		name string
+		urlA string
+		urlB string
+		want bool
 	}{
-		{"2nd level", "quay.io", "quay.io"},
-		{"3nd level", "sub.quay.io", "quay.io"},
-		{"localhost", "localhost", "localhost"},
-		{"2nd level with protocol", "https://docker.io", "docker.io"},
-		{"2nd level with path", "docker.io/v1/", "docker.io"},
-		{"2nd level with port", "docker.io:80", "docker.io"},
-		{"2nd level with protocol and path", "https://docker.io/v1/", "docker.io"},
-		{"3rd level with protocol and path", "https://index.docker.io/v1/", "docker.io"},
-		{"3rd level with protocol and path and port", "https://index.docker.io:80/v1/", "docker.io"},
-		{"localhost with protocol and path and port", "http://localhost:8080/v1/", "localhost"},
+		{"no port matching host", "quay.io", "quay.io", true},
+		{"non-matching host different sub-domain", "sub.quay.io", "quay.io", false},
```

Comment:
Really small nitpick here, but maybe make this descriptor slightly different than the one below where there are two different subdomains.
```suggestion
		{"non-matching host added sub-domain", "sub.quay.io", "quay.io", false},
```